src/main/java/hudson/plugins/git/ChangelogToBranchOptions.java

```diff
@@ -12,7 +12,7 @@
  *
  * @author <a href="mailto:dirk.reske@t-systems.com">Dirk Reske (dirk.reske@t-systems.com)</a>
  */
-public class ChangelogToBranchOptions extends AbstractDescribableImpl<ChangelogToBranchOptions> implements Serializable {
+public class ChangelogToBranchOptions extends AbstractDescribableImpl<ChangelogToBranchOptions> implements ChangelogOptions, Serializable {
```

Comment:
Does the change in class definition require a new serial version ID, or other changes to handle migration and serialization compatibility?

Reply:
I think if we only add a new interface that wouldn't be a problem would it? I'm not super familiar with serial versions.

Reply:
Ok, so I read more on this. The object serialization is described in https://docs.oracle.com/javase/9/docs/specs/serialization/version.html, and defines what incompatible and compatible changes are:

Incompatible changes are the following, 

5.6.1 Incompatible Changes

Incompatible changes to classes are those changes for which the guarantee of interoperability cannot be maintained. The incompatible changes that may occur while evolving a class are:

Deleting fields - If a field is deleted in a class, the stream written will not contain its value. When the stream is read by an earlier class, the value of the field will be set to the default value because no value is available in the stream. However, this default value may adversely impair the ability of the earlier version to fulfill its contract.

Moving classes up or down the hierarchy - This cannot be allowed since the data in the stream appears in the wrong sequence.

Changing a nonstatic field to static or a nontransient field to transient - When relying on default serialization, this change is equivalent to deleting a field from the class. This version of the class will not write that data to the stream, so it will not be available to be read by earlier versions of the class. As when deleting a field, the field of the earlier version will be initialized to the default value, which can cause the class to fail in unexpected ways.

Changing the declared type of a primitive field - Each version of the class writes the data with its declared type. Earlier versions of the class attempting to read the field will fail because the type of the data in the stream does not match the type of the field.

Changing the writeObject or readObject method so that it no longer writes or reads the default field data or changing it so that it attempts to write it or read it when the previous version did not. The default field data must consistently either appear or not appear in the stream.

Changing a class from Serializable to Externalizable or vice versa is an incompatible change since the stream will contain data that is incompatible with the implementation of the available class.

Changing a class from a non-enum type to an enum type or vice versa since the stream will contain data that is incompatible with the implementation of the available class.

Removing either Serializable or Externalizable is an incompatible change since when written it will no longer supply the fields needed by older versions of the class.

Adding the writeReplace or readResolve method to a class is incompatible if the behavior would produce an object that is incompatible with any older version of the class.

Since this change to the class only changes the implemented interfaces and (after i make the requested changes) will only add a new method, and does not change any fields, this should remain compatible. Note also that no serial ID has been defined for the class. Perhaps we should do so now....